src/pybind/mgr/dashboard/frontend/src/app/shared/directives/auth-storage.directive.ts

```diff
@@ -0,0 +1,93 @@
+import { Directive, Input, TemplateRef, ViewContainerRef } from '@angular/core';
+
+import _ from 'lodash';
+
+import { Permission, Permissions } from '~/app/shared/models/permissions';
+import { AuthStorageService } from '~/app/shared/services/auth-storage.service';
+
+@Directive({
+  selector: '[cdAuthStorage]'
+})
+export class AuthStorageDirective {
+  permissions: Permissions;
+
+  constructor(
+    private templateRef: TemplateRef<any>,
+    private viewContainer: ViewContainerRef,
+    private authStorageService: AuthStorageService
+  ) {}
+
+  @Input('cdAuthStorage') set cdAuthStorage(condition: string | string[] | object) {
+    this.permissions = this.authStorageService.getPermissions();
+    if (this.isAuthorized(condition)) {
+      this.viewContainer.createEmbeddedView(this.templateRef);
+    } else {
+      this.viewContainer.clear();
+    }
+  }
+
+  private isAuthorized(condition: any): boolean {
+    /*
+      single permission equivalent to check e.g osd.
+      We check if the permission key exists on the Permissions class then
+      if true We check for read permission and display the component accordingly
+    */
+
+    if (_.isString(condition)) {
+      if (_.has(this.permissions, [condition, 'read']) && this.permissions[condition]['read']) {
+        return true;
+      }
```

Comment:
This is just a trick: lodash gives you the [`_.get(obj, path[], default)`](https://lodash.com/docs/4.17.15#get) method with a default value for the case the path is not found in the object:

```suggestion
    return _.get(this.permissions, [condition, 'read'], false);
```